lib/utils/streams.ts

```diff
@@ -0,0 +1,20 @@
+import { ReadStream } from 'fs';
+
+export async function buffer(stream: NodeJS.ReadableStream, bufferFile: string) {
+	const Promise = await import('bluebird');
+	const fs = await import('fs');
+
+	const fileWriteStream = fs.createWriteStream(bufferFile);
+
+	return new Promise(function(resolve, reject) {
+		return stream
+		.on('error', reject)
+		.on('end', resolve)
+		.pipe(fileWriteStream);
+	}).then(() => new Promise(function(resolve, reject) {
+		fs.createReadStream(bufferFile)
+		.on('open', function(this: ReadStream) {
```

Comment:
Could this be `.on('open', resolve)` and use `new Promise<ReadStream>`?
Actually I just tried that and got `Promise<ReadStream>` instead of `Promise<{}>` inferred as the return type of the function.

Reply:
> Could this be .on('open', resolve)

Nope, I think the `this` argument isn't doing what you think here. That doesn't mean that there's a first argument called `this`. It's just telling typescript the implicit `this` that the function will be called with. This code is equivalent to:

```js
.on('open', function () {
  resolve(this);
})
```

It's a bit tricky: we're basically saying 'when the stream opens, resolve with the stream itself', without having an actual stream variable flying around anywhere. You're totally right though, it was confusingly written, I've simplified it.

Reply:
Ohhhh, :sparkles: :sparkling_heart: :heart_eyes:

Reply:
Is there a reason that the `ReadStream` isn't used any more? The `buffer()` now returns `Promise<{}>`. Should we use `new Promise<ReadStream>` or similar casting to fix that?